packages/core/client.js

```diff
@@ -298,6 +302,8 @@ class Client {
 
       if (!shouldSend) {
         this._logger.debug('Event not sent due to onError callback')
+        trackSession(event)
```

Comment:
I'm tempted to say that this is unnecessary too, but the spec is unclear so I've started an internal discussion to get clarity.